src/channel_internal.c

```diff
@@ -139,17 +151,29 @@ static void execute_callbacks(void* context)
         CHANNEL_OP* channel_op = (CHANNEL_OP*)context;
         CHANNEL_CALLBACK_RESULT result = channel_op->result; // local copy to make sure both callbacks are called with the same result
 
+        LOGGER_LOG(LOG_LEVEL_VERBOSE, T_PTR_VALUE_OR_NULL(channel_op->channel_internal->log_context),
+            "Executing callbacks for"
+            " pull_correlation_id = %" PRI_RC_STRING ""
+            ", push_correlation_id = %" PRI_RC_STRING ""
+            ", callback_result=%" PRI_MU_ENUM "",
+            RC_STRING_VALUE_OR_NULL(channel_op->pull_correlation_id),
+            RC_STRING_VALUE_OR_NULL(channel_op->push_correlation_id),
+            MU_ENUM_VALUE(CHANNEL_CALLBACK_RESULT, result)
+        );
+
         /*Codes_SRS_CHANNEL_INTERNAL_43_145: [ execute_callbacks shall call the stored callback(s) with the result of the operation. ]*/
         if (channel_op->pull_callback)
         {
-            channel_op->pull_callback(channel_op->pull_context, result, channel_op->data);
+            channel_op->pull_callback(channel_op->pull_context, result, channel_op->pull_correlation_id, channel_op->push_correlation_id, channel_op->data);
```

Comment:
> channel_op->push_correlation_id [](http://example.com/codeflow?start=104&length=32)

Why is `push_correlation_id` used when the condition in line 165 (two lines above) is `if (channel_op->pull_callback)` (note: `pull_callback`).

Q1: Is this intended? 
Q2: if it was not intended, how was it tested?

Reply:
Yes this is intended. 

The `if(channel_op->pull_callback` checks if the `pull_callback` field was set for this operation.

The `push_correlation_id` is being passed to the callback so that the user knows which `channel_push` caused this specific callback to trigger.